api/v1alpha3/azurecluster_webhook.go

```diff
@@ -29,3 +35,40 @@ func (r *AzureCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		For(r).
 		Complete()
 }
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-azurecluster,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azurecluster,versions=v1alpha3,name=validation.azurecluster.infrastructure.cluster.x-k8s.io
+
+var _ webhook.Validator = &AzureCluster{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *AzureCluster) ValidateCreate() error {
+	machinelog.Info("validate create", "name", r.Name)
+
+	return nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *AzureCluster) ValidateUpdate(old runtime.Object) error {
+	machinelog.Info("validate update", "name", r.Name)
+	azureCluster, ok := old.(*AzureCluster)
+	if !ok {
+		return fmt.Errorf("update object is not a AzureCluster type")
+	}
+
+	if azureCluster.Spec.Location != r.Spec.Location {
```

Comment:
I think Location was just an example, but there are more fields that need validation

Reply:
let's focus on validating Location for now as a pretext to add the necessary foundation for AzureCluster validation and open follow up issues for other validations (this one was a good first issue)

Reply:
sounds good!